client/src/modules/journal/modals/search.modal.js

```diff
@@ -22,11 +27,30 @@ function JournalSearchModalController(Instance, Projects, Notify, Store, filters
   // assign already defined custom filters to searchQueries object
   vm.searchQueries = util.maskObjectFromKeys(filters, searchQueryOptions);
 
+  // has default account
+  if (options.hasDefaultAccount) {
```

Comment:
It would be useful to document the behaviour of `hasDefaultAccount` somewhere in a comment. The branching logic here could be difficult to follow if you didn't see that it was used by different modules with different needs.